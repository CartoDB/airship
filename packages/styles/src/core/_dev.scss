@import 'tools/_mixins';
@import 'variables/_variables';
