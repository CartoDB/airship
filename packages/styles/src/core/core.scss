@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
