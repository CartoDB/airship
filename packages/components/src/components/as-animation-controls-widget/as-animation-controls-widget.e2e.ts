import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-animation-controls-widget', () => {
  describe('Rendering', () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage();

      await page.setContent(`
        <as-animation-controls-widget
          heading='Title'
          description='Description'
          >
        </as-animation-controls-widget>
      `);
    });

    it('should render properly', async () => {
      const element = await page.find('as-animation-controls-widget');

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render the range slider thumb if set to true', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('showThumb', true);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render show the range slider thumb if set to false', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('showThumb', false);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render show the range slider thumb caption if set to true', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('showCaption', true);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render show the range slider thumb caption if set to false', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('showCaption', false);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render the widget header if showHeader property is False', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('showHeader', false);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render the header if the heading is not present', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('heading', null);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render the header if the description is not present', async () => {
      const element = await page.find('as-animation-controls-widget');
      element.setProperty('description', null);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('Behaviour', () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage();

      await page.setContent(`
        <as-animation-controls-widget></as-animation-controls-widget>
      `);
    });

    it('should emit a play event when clicking to the button the first time', async () => {
      const element: E2EElement = await page.find('as-animation-controls-widget');
      const playPauseButton = await element.find('.as-animation-controls-widget__player > button');
      const pauseEventSpy = await element.spyOnEvent('play');

      // Click on pause
      playPauseButton.click();
      await page.waitForChanges();

      expect(pauseEventSpy).toHaveReceivedEvent();
    });

    it('should emit a pause event when clicking to the button the second time', async () => {
      const element: E2EElement = await page.find('as-animation-controls-widget');
      const playPauseButton = await element.find('.as-animation-controls-widget__player > button');
      const playEventSpy = await element.spyOnEvent('pause');

      // Click on play and pause
      playPauseButton.click();
      playPauseButton.click();
      await page.waitForChanges();

      expect(playEventSpy).toHaveReceivedEvent();
    });
  });
});
