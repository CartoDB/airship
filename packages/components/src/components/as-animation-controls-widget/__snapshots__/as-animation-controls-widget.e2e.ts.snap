// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-animation-controls-widget Rendering should not render show the range slider thumb caption if set to false 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should not render show the range slider thumb if set to false 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should not render the header if the description is not present 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\"></p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should not render the header if the heading is not present 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\"></h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should not render the widget header if showHeader property is False 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should render properly 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should render show the range slider thumb caption if set to true 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;

exports[`as-animation-controls-widget Rendering should render the range slider thumb if set to true 1`] = `"<as-animation-controls-widget heading=\\"Title\\" description=\\"Description\\" class=\\"hydrated\\"> <as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Title</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-animation-controls-widget__wrapper as-body\\"><div class=\\"as-animation-controls-widget__player\\"><button class=\\"as-btn\\"><svg width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"var(--as--color--primary)\\" d=\\"M8,15 C4.1402,15 1,11.8605 1,8 C1,4.1409 4.1402,1 8,1 C11.8598,1 15,4.1402 15,8 C15,11.8605 11.8598,15 8,15 Z M8,2.4 C4.9123,2.4 2.4,4.9123 2.4,8 C2.4,11.0877 4.9123,13.6 8,13.6 C11.0877,13.6 13.6,11.0877 13.6,8 C13.6,4.9123 11.0877,2.4 8,2.4 Z M6.5,10.733336 L6.5,5.4 L10.500002,8.066668 L6.5,10.733336 Z\\"></path></svg></button><as-range-slider min-value=\\"0\\" max-value=\\"100\\" class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\" style=\\"left: 0%;\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--hidden\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\" style=\\"left: 0%; width: 0%;\\"></div></as-range-slider-bar></div></div></as-range-slider></div><footer class=\\"as-animation-controls__progress-value\\"></footer></div></as-animation-controls-widget>"`;
